### PR TITLE
Change OnPreparingSpawnable notification to be specific to tools

### PR DESCRIPTION
**Pull request by michabr**
The "OnPreparingSpawnable" notification was a generic notification that was being called on every spawnable load. This PR replaces this notification with a more specific one that is only called on the tools side when a prefab template is to be processed and converted into an in-memory spawnable. This is used in editor networking code so that unchanged spawnables can be sent to the server and have their aliases resolved there instead of resolving them first on the client, which may have different alias resolving logic.

Signed-off-by: abrmich <abrmich@amazon.com>

### CODE REVIEW

File: Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Spawnable/PrefabToInMemorySpawnableNotificationBus.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#include <AzCore/EBus/EBus.h>
+#include <AzFramework/Entity/EntityContext.h>
+
+namespace AzToolsFramework::Prefab
+{
+    //! This EBus is used to send notifications during the conversion of a prefab template to an in-memory spawnable
+    class PrefabToInMemorySpawnableNotifications
+        : public AZ::EBusTraits
+    {
+    public:
```
Comment (NicholasVanSickle) [inline Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Spawnable/PrefabToInMemorySpawnableNotificationBus.h:20]: nit, suggest making bus policies explicit for clarity:
```
        static constexpr EBusHandlerPolicy HandlerPolicy = EBusHandlerPolicy::Multiple;
        static constexpr EBusAddressPolicy AddressPolicy = EBusAddressPolicy::Single;
```

File: Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableAssetHandler.cpp [hunk 2]
```diff
@@ -55,7 +54,6 @@ namespace AzFramework
         AZ::ObjectStream::FilterDescriptor filter(assetLoadFilterCB);
         if (AZ::Utils::LoadObjectFromStreamInPlace(*stream, *spawnable, nullptr /*SerializeContext*/, filter))
         {
-            SpawnableAssetEventsBus::Broadcast(&SpawnableAssetEventsBus::Events::OnPreparingSpawnable, *spawnable, asset.GetHint());
```
Comment (NicholasVanSickle) [inline Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableAssetHandler.cpp:58]: should this bus be removed / deprecated or is this called elsewhere?

**Review by NicholasVanSickle — APPROVED** [review]

File: Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableAssetHandler.cpp (hunk 2 above)
Comment (michabr) [reply Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableAssetHandler.cpp:58]: This bus already existed and contains other events that are still being used, so it's still being called elsewhere.

File: Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Spawnable/PrefabToInMemorySpawnableNotificationBus.h (hunk 1 above)
Comment (michabr) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Spawnable/PrefabToInMemorySpawnableNotificationBus.h:20]: Done.

File: Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Spawnable/PrefabToInMemorySpawnableNotificationBus.h [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+#pragma once
+
+#include <AzCore/EBus/EBus.h>
+#include <AzFramework/Entity/EntityContext.h>
+
+namespace AzToolsFramework::Prefab
+{
+    //! This EBus is used to send notifications during the conversion of a prefab template to an in-memory spawnable
+    class PrefabToInMemorySpawnableNotifications
+        : public AZ::EBusTraits
```
Comment (AMZN-alexpete) [inline Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Spawnable/PrefabToInMemorySpawnableNotificationBus.h:18]: This bus is not protected by a recursive mutex like the previous one - are there potential concerns with threading here?

Comment (michabr) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Spawnable/PrefabToInMemorySpawnableNotificationBus.h:18]: That's a good callout but the recursive mutex should no longer be necessary as this replacement notification is smaller in scope and should only be called on the main thread.

**Review by AMZN-alexpete — APPROVED** [review]

**Pull request merged**